src/main/java/org/ow2/proactive/catalog/repository/CatalogObjectRepository.java

```diff
@@ -52,9 +52,15 @@
     @Query(value = "SELECT DISTINCT cos.contentType FROM CatalogObjectEntity cos")
     Set<String> findAllContentTypes();
 
+    @Query(value = "SELECT cos.bucket.bucketName, cos.id.name FROM CatalogObjectEntity cos WHERE lower(cos.kind) LIKE lower(concat('%', ?1, '%')) AND lower(cos.contentType) LIKE lower(concat('%', ?2, '%'))")
+    List<Object[]> findCatalogObjectNameReferenceByKindAndContentType(String kind, String contentType);
+
     @Query(value = "SELECT cos.bucket.bucketName, cos.id.name FROM CatalogObjectEntity cos WHERE lower(cos.kind) LIKE lower(concat('%', ?1, '%'))")
     List<Object[]> findCatalogObjectNameReferenceByKind(String kind);
 
+    @Query(value = "SELECT cos.bucket.bucketName, cos.id.name FROM CatalogObjectEntity cos WHERE lower(cos.contentType) LIKE lower(concat('%', ?1, '%'))")
+    List<Object[]> findCatalogObjectNameReferenceByContentType(String contentType);
```

Comment:
this method can be removed if you will use just general version with kind = ""
findCatalogObjectNameReferenceByKindAndContentType

Reply:
@ShatalovYaroslav I don't know how spring works but in SQL if you have a where clause with `kind=""` then you require a join that satisfies this condition, which will always return 0 results as all catalog objects have a kind.

Reply:
@aminelouati says it works so I guess Spring does some auto-mapping and transforms kind="" to kind="*"

Reply:
Nice catch @ShatalovYaroslav  ;)